pkg/resources/grant_helpers.go

```diff
@@ -229,15 +229,29 @@ func readGenericGrant(
 		}
 		return err
 	}
+
 	priv := d.Get("privilege").(string)
 	grantOption := d.Get("with_grant_option").(bool)
 
+	var relevantGrants []*grant
+	for _, grant := range grants {
+		if grant.Privilege == priv && grant.GrantOption == grantOption {
+			relevantGrants = append(relevantGrants, grant)
+		}
+	}
+
+	// If no relevant grants, set id to blank and return (see HACK HACK comment above)
+	if len(relevantGrants) == 0 {
```

Comment:
I thought that returning empty ID is the hack. (from the comment above that mentions the HACK). Regardless, I check is OK. There is no error code in this case. A grant (or grants were) was found but non relevant to our search since they do not match the privilege or grant option properties.